modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Actions.tsx

```diff
@@ -68,19 +70,53 @@ const TYPES = [
 
 const noop = () => {};
 
-const Actions = () => {
+const Actions = ({spritemap}: {spritemap: string}) => {
 	const [activeSection, setActiveSection] = useState(SECTIONS.ACTIONS);
 	const [activeTab, setActiveTab] = useState(0);
+	const [availableIconSymbols, setAvailableIconSymbols] = useState<
+		Array<{label: string; value: string}>
+	>([]);
 	const [
 		confirmationMessageTranslations,
 		setConfirmationMessageTranslations,
 	] = useState({});
 	const [iconSymbol, setIconSymbol] = useState('bolt');
 	const [labelTranslations, setLabelTranslations] = useState({});
 
+	useEffect(() => {
+		const getIcons = async () => {
+			const response = await fetch(spritemap);
+
+			const responseText = await response.text();
+
+			if (responseText.length) {
+				const XMLString = await new window.DOMParser().parseFromString(
+					responseText,
+					'text/xml'
+				);
+
+				const availableIconSymbolElements = XMLString.querySelectorAll(
+					'symbol'
+				);
```

Comment:
Interesting way of parsing. Pure string parsing, with a regex, would probably have better performance, but I think a small performance hit is worth for the simplification in code. :+1: 

Is `await` needed?

nits:
- `XMLString` should normally be lowercase, `xmlString`. But in this case, the variable does not contain a string value. It should be something like `spritemapDocument`.
- `window` is not necessary
- `availableIconSymbolElements` can be just `symbolElements`